### PR TITLE
Add view capturing capability

### LINKED ISSUE #224: Add ViewCapture Method
#### Description:

Add a method to capture the active viewport. 

Rhino command `-ViewCaptureToFile` with some standard settings baked in such as width/height and scale.

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #224

Add capability to capture rhino view.
- Method for capturing currently active view
- Method for capturing all named views

Note that this requires an uptick in the nugets to Rhino 6, as the ViewCapture class used is internal in rhino 5. There are ways around this for rhino 5 as well, but from testing them, they perform significantly worse and does not give the same level of control.

Before this PR is merged an agreement that this uptick is ok, as it means no more support for rhino 5 (as in, might still work, but can not be as relied upon any longer). @al-fisher @FraserGreenroyd , have discussed this with you before raising this, but noting you in here as well.

### CODE REVIEW

**Review by michaelhoehn — COMMENTED** [review]
Really nice and easy to use! It took a minute to understand the distinction between a DimensionView and a ScaleView since this functionality is unified in a standard Rhino command line prompt, but I think I'm good with the approach generally. I did notice however that there are some performance quirks when using one or the other when exporting a sample image 5000x5000 (likely larger than standard outputs, but I noticed similar behaviors with anything over 3K). 
image
This is entirely not in our perview as I see it, but good to simply document here it seems capture methodology could have varying nuances between the different types (dimensioned on the left, and scaled on the right). 

I'm happy to provide an approving review when needed, but as noted in the PR and in my comments, we need to discuss our decision to leave Rhino 5 behind in support of these newer features.

File: Rhinoceros_Engine/packages.config [hunk 1]
```diff
@@ -1,4 +1,4 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="RhinoCommon" version="5.12.50810.13095" targetFramework="net452" />
+  <package id="RhinoCommon" version="6.33.20343.16431" targetFramework="net472" />
```
[the remark this answers is not in the document]
Comment (michaelhoehn) [reply Rhinoceros_Engine/packages.config:3]: https://github.com/BHoM/Grasshopper_UI/issues/672

**Review by michaelhoehn — DISMISSED** [review]
Very happy with this implementation!

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by michaelhoehn** [comment]
@BHoMBot check compliance

**Comment by michaelhoehn** [comment]
@BHoMBot fix copyright headers ref. 10441140689

**Comment by FraserGreenroyd** [comment]
@BHoMBot fix copyright headers ref. 10441140689

File: Rhinoceros_Engine/Compute/CaptureNamedViews.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline Rhinoceros_Engine/Compute/CaptureNamedViews.cs:3]:
```suggestion
 * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
```

File: Rhinoceros_Engine/Compute/CaptureView.cs [hunk 3]
```diff
@@ -0,0 +1,205 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline Rhinoceros_Engine/Compute/CaptureView.cs:3]:
```suggestion
 * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
```

File: Rhinoceros_oM/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline Rhinoceros_oM/Properties/AssemblyInfo.cs:3]:
```suggestion
 * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
```

File: Rhinoceros_oM/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+//
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("5.0.0.0")]
```
Comment (FraserGreenroyd) [inline Rhinoceros_oM/Properties/AssemblyInfo.cs:56]:
```suggestion
[assembly: AssemblyVersion("6.0.0.0")]
```

File: Rhinoceros_oM/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+[assembly: AssemblyFileVersion("5.3.0.0")]
```
Comment (FraserGreenroyd) [inline Rhinoceros_oM/Properties/AssemblyInfo.cs:57]:
```suggestion
[assembly: AssemblyFileVersion("6.1.0.0")]
```

File: Rhinoceros_oM/ViewCapture/DimensionViewCaptureSettings.cs [hunk 7]
```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline Rhinoceros_oM/ViewCapture/DimensionViewCaptureSettings.cs:3]:
```suggestion
 * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
```

File: Rhinoceros_oM/ViewCapture/IViewCaptureSettings.cs [hunk 8]
```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline Rhinoceros_oM/ViewCapture/IViewCaptureSettings.cs:3]:
```suggestion
 * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
```

File: Rhinoceros_oM/ViewCapture/ScaleViewCaptureSettings.cs [hunk 9]
```diff
@@ -0,0 +1,67 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
```
Comment (FraserGreenroyd) [inline Rhinoceros_oM/ViewCapture/ScaleViewCaptureSettings.cs:3]:
```suggestion
 * Copyright (c) 2015 - 2023, the respective contributors. All rights reserved.
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by michaelhoehn** [comment]
@BHoMBot check compliance

File: Rhinoceros_oM/ViewCapture/ScaleViewCaptureSettings.cs [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    [Description("View capture settings allowing the size of the image to be controled as a scale of the current viewport size.")]
+    public class ScaleViewCaptureSettings : IViewCaptureSettings
+    {
+
+        /***************************************************/
+        /**** Properties                                ****/
+        /***************************************************/
+
+        [Description("Scale factor of the current preview. A factor of 2 gives a resolution twice to the current viewport resolution.")]
+        public double Scale { get; set; } = 2.0;
```
Comment (FraserGreenroyd) [inline Rhinoceros_oM/ViewCapture/ScaleViewCaptureSettings.cs:42]:
```suggestion
        public virtual double Scale { get; set; } = 2.0;
```

**Review by FraserGreenroyd — CHANGES_REQUESTED** [review]

**Comment by michaelhoehn** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check required

**Review by michaelhoehn — APPROVED** [review]
Go for merge!

**Comment by IsakNaslundBh** [comment]
@BHoMBot check null-handling

**Comment by FraserGreenroyd** [comment]
@BHoMBot check null-handling

**Comment by FraserGreenroyd** [comment]
@BHoMBot check compliance

**Comment by FraserGreenroyd** [comment]
@BHoMBot check core
@BHoMBot check serialisation
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
I think I've figured out why versioning is failing - installer would too if we ran it - LadybugTools_Toolkit has a dependency on this toolkit, I think this PR is breaking LadybugTools_Toolkit's ability to compile, thus resulting in versioning issues when the methods/objects don't exist.

I will take another look when back on the laptop to see if I can compile LadybugTools_Toolkit against this branch and if not, work out what the rectification might need to be early in the week with @tg359

**Comment by FraserGreenroyd** [comment]
@BHoMBot check versioning

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Review by FraserGreenroyd — APPROVED** [review]
Now that checks are passing, approving based on previous reviews.

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 10455686053

**Comment by IsakNaslundBh** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**